org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DiagnosticsHandler.java

```diff
@@ -146,22 +150,26 @@ private static DiagnosticSeverity convertSeverity(IProblem problem) {
 	}
 
 	@SuppressWarnings("restriction")
-	private static Range convertRange(IProblem problem) {
-		Position start = new Position();
-		Position end = new Position();
-
-		start.setLine(problem.getSourceLineNumber()-1);// VSCode is 0-based
-		end.setLine(problem.getSourceLineNumber()-1);
-		if(problem instanceof DefaultProblem){
-			DefaultProblem dProblem = (DefaultProblem)problem;
-			start.setCharacter(dProblem.getSourceColumnNumber() - 1);
-			int offset = 0;
-			if (dProblem.getSourceStart() != -1 && dProblem.getSourceEnd() != -1) {
-				offset = dProblem.getSourceEnd() - dProblem.getSourceStart() + 1;
+	private static Range convertRange(IOpenable openable, IProblem problem) {
+		try {
+			return JDTUtils.toRange(openable, problem.getSourceStart(), problem.getSourceEnd() - problem.getSourceStart() + 1);
+		} catch (CoreException e) {

```

Comment:
in which case are we expected to catch CoreExceptions?

Reply:
When try to access the Buffer of the IOpenable obect, it will throw JavaModelException. In this case, just use the previous behavior, which only based on the information of IProblem.

Reply:
ok please add a comment in the code then